Core/HLE/proAdhoc.cpp

```diff
@@ -1896,13 +1896,35 @@ uint16_t getLocalPort(int sock) {
 	return ntohs(localAddr.sin_port);
 }
 
-u_long getAvailToRecv(int sock) {
+u_long getAvailToRecv(int sock, int udpBufferSize) {
 	u_long n = 0; // Typical MTU size is 1500
+	int err = -1;
 #if defined(_WIN32) // May not be available on all platform
-	ioctlsocket(sock, FIONREAD, &n);
+	err = ioctlsocket(sock, FIONREAD, &n);
 #else
-	ioctl(sock, FIONREAD, &n);
+	err = ioctl(sock, FIONREAD, &n);
 #endif
+	if (err < 0)
+		return 0;
+
+	if (udpBufferSize > 0 && n > 0) {
+		// Cap number of bytes of full DGRAM message(s?) up to buffer size
+		static int lastUdpBufSize = 0;
+		static char* buf = NULL;
+		if (udpBufferSize > lastUdpBufSize) {
+			// Reusing temp buffer to prevent causing too many fragmentation due to repeated alloc -> free (was getting out of memory issue)
```

Comment:
if you were really running out of memory, you would have been leaking. I don't think fragmentation here would cause such serious issues..

Reply:
It might not be fragmentation, i only gets out of memory when debugging it through Visual Studio 2019, may be it's VS bug on memory allocation?
It's just a simple:
```c++
char *buf = (char*)malloc(size);
err = recvfrom(...);
free(buf);
```
So i can't really tell where the leaks came from, and i certainly don't want to get issue while debugging.

Reply:
That's really odd, something like that shouldn't run of RAM.

However, if realloc fixes it, let's go with that, I guess..

Reply:
Hmm... i think we don't really need to provide a large buffer since we only need to retrieved the message size, and it seems with `MSG_TRUNC` we can provide small buffer and it will still return the actual message size (one message)